pkg/apis/policy/types.go

```diff
@@ -352,6 +364,19 @@ const (
 	RunAsUserStrategyRunAsAny RunAsUserStrategy = "RunAsAny"
 )
 
+// RunAsGroupStrategy denotes strategy types for generating RunAsGroup values for a
+// SecurityContext.
+type RunAsGroupStrategy string
+
+const (
+	// RunAsGroupStrategyMustRunAs means that container must run as a particular uid.
+	RunAsGroupStrategyMustRunAs RunAsGroupStrategy = "MustRunAs"
+	// RunAsGroupStrategyMustRunAsNonRoot means that container must run as a non-root uid
+	RunAsGroupStrategyMustRunAsNonRoot RunAsGroupStrategy = "MustRunAsNonRoot"
```

Comment:
drop this one until https://github.com/kubernetes/kubernetes/pull/62216 is decided on?

Reply:
Do we even need this? Isn't it the same as:
```yaml
runAsGroup:
  rule: MustRunAs
  ranges:
    - min: 1
      max: 2147483647
```

Reply:
@liggitt   i think #62216  is mostly about not needing NonRootGroup in Pod Spec. I think there is agreement that we need this feature at PSP level.
@tallclair agree MustRunAs can achieve that. It would be just inconsistent with how we do non root for RunAsUser.

its basically a choice between what you posted vs this below. More about usability. I think just from usability and inconsistency point of view we should stick with the bool
```
runAsGroup:
  runAsNonRoot: true
```

Reply:
Until we have a dedicated runAsNonRootGroup field in the pod spec, I'd prefer to model the PSP restriction as Tim described above. The only reason we have a runAsNonRoot for user in PSP is because it is controlling a corresponding field in the pod spec